server/pulp/server/db/model/criteria.py

```diff
@@ -177,6 +177,46 @@ def __init__(self, type_ids=None, association_filters=None, unit_filters=None,
 
         self.remove_duplicates = remove_duplicates
 
+    def to_dict(self):
+        """
+        :return:    the UnitAssociationCriteria as a dict, suitable for serialization by
+                    something like JSON, and compatible as input to the from_dict method.
+        :rtype:     dict
+        """
+
+        return {
+            'type_ids': self.type_ids,
+            'association_filters': self.association_filters,
+            'unit_filters': self.unit_filters,
+            'association_sort': self.association_sort,
+            'unit_sort': self.unit_sort,
+            'limit': self.limit,
+            'skip': self.skip,
+            'association_fields': self.association_fields,
+            'unit_fields': self.unit_fields,
+            'remove_duplicates': self.remove_duplicates
+        }
+
+    @classmethod
+    def from_dict(cls, input_dictionary):
+        """
+        Convert a dictionary representation of the UnitAssociationCriteria into a new
+        UnitAssociationCriteria object.
+        The output of to_dict() is suitable as input to this method.
```

Comment:
Consider adding a newline above this to separate it from the paragraph above it.